app/controllers/api/generic_events_controller.rb

```diff
@@ -0,0 +1,56 @@
+module Api
+  class GenericEventsController < ApiController
+    PERMITTED_EVENT_PARAMS = [
+      :type,
+      attributes: [
+        :event_type,
+        :occurred_at,
+        :recorded_at,
+        :notes,
+        details: {},
+      ],
+      relationships: {},
+    ].freeze
+
+    def create
+      GenericEvents::CommonParamsValidator.new(event_params).validate!
+      event = event_type.constantize.create(event_attributes)
```

Comment:
can you use `create!` to fail on validation errors

Reply:
what is the plan with triggering the event actions and persisting updates on eventables, as this only creates the event but doesn't cancel or update the move?

Reply:
Coming later

Reply:
Yeah, makes sense. I was following the idiom we've got currently. But that's clearly wrong.